lib/Runtime/Library/ForInObjectEnumerator.cpp

```diff
@@ -196,7 +197,11 @@ namespace Js
                     propertyId = propRecord->GetPropertyId();
                 }
 
-                if (TestAndSetEnumerated(propertyId) //checks if the property is already enumerated or not
+                if (!this->shadowData->initialObject->IsEnumerable(propertyId))
```

Comment:
I am wondering if this is the right fix.
Why are non-enumerable properties added with `Object.defineProperty(obj, "p", {value: 1, enumerable: false})` do not show up today in for-on loop ? We should make sure that `length` follows the same logic